transformer_engine/pytorch/dot_product_attention/rope.py

```diff
@@ -119,13 +119,21 @@ def forward(
         ctx,
         t: torch.Tensor,
         freqs: torch.Tensor,
+        start_positions: Union[torch.Tensor, None] = None,
         tensor_format: str = "sbhd",
         interleaved: bool = False,
         cu_seqlens: Union[torch.Tensor, None] = None,
         cp_size: int = 1,
         cp_rank: int = 0,
     ) -> torch.Tensor:
         """Fused RoPE forward."""
+
+        if start_positions is None:
```

Comment:
How about wrap `start_positions` with `std::optional` in the extension interface, so we don't need to create an empty tensor every time? We did observe some CPU overheads of it before.